From 8d0a517e5a06a2737ca0b8a83e9cdfc2f2aa04a8 Mon Sep 17 00:00:00 2001
From: Wolfgang Grandegger <wg@aries-embedded.de>
Date: Fri, 13 Feb 2026 18:27:14 +0100
Subject: [PATCH] MSRZV2H board configuration

Add support for the ARIES Embedded MSRZV2H modules on the
V2Hberry base board.

Signed-off-by: Wolfgang Grandegger <wg@aries-embedded.de>
---
 arch/arm/dts/Makefile               |   1 +
 arch/arm/dts/msrzv2hberry.dts       | 735 ++++++++++++++++++++++++++++
 arch/arm/mach-renesas/Kconfig.rcar3 |   6 +
 board/aries/msrzv2h/Kconfig         |  15 +
 board/aries/msrzv2h/Makefile        |  13 +
 board/aries/msrzv2h/msrzv2h.c       | 324 ++++++++++++
 configs/msrzv2hberry_defconfig      | 189 +++++++
 include/configs/msrzv2hberry.h      |  68 +++
 8 files changed, 1351 insertions(+)
 create mode 100644 arch/arm/dts/msrzv2hberry.dts
 create mode 100644 board/aries/msrzv2h/Kconfig
 create mode 100644 board/aries/msrzv2h/Makefile
 create mode 100644 board/aries/msrzv2h/msrzv2h.c
 create mode 100644 configs/msrzv2hberry_defconfig
 create mode 100644 include/configs/msrzv2hberry.h

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 6048d18c65..681f999307 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -966,6 +966,7 @@ dtb-$(CONFIG_RCAR_GEN3) += \
 	rzg2ul-type1-ddr3l-dev.dtb \
 	rzg2ul-type2-dev.dtb \
 	smarc-rzg2ul.dtb \
+	msrzv2hberry.dtb \
 	rzv2h-dev.dtb \
 	rzv2h-evk-alpha.dtb \
 	rzv2h-evk-ver1.dtb \
diff --git a/arch/arm/dts/msrzv2hberry.dts b/arch/arm/dts/msrzv2hberry.dts
new file mode 100644
index 0000000000..df965c8858
--- /dev/null
+++ b/arch/arm/dts/msrzv2hberry.dts
@@ -0,0 +1,735 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source for the RZ/V2H EVK ver1 board
+ *
+ * Copyright (C) 2024 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "r9a09g057.dtsi"
+#include <dt-bindings/pinctrl/rzg2l-pinctrl.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	model = "ARIES Embedded MSRZV2H module on V2Hberry base board";
+	compatible = "aries,msrzv2hberry", "renesas,r9a09g057h4", "renesas,r9a09g057";
+
+	aliases {
+		serial0 = &scif0;
+		serial1 = &sci0;
+		serial2 = &sci4;
+		serial3 = &sci5;
+		mmc0 = &sdhi0;
+		mmc1 = &sdhi1;
+
+		ethernet0 = &eth0;
+		ethernet1 = &eth1;
+	};
+
+	chosen {
+		bootargs = "ignore_loglevel";
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory@48000000 {
+		device_type = "memory";
+		/* first 128MB is reserved for secure area. */
+		reg = <0x0 0x48000000 0x1 0xF8000000>;
+	};
+
+	//memory@240000000 {
+	//	device_type = "memory";
+	//	reg = <0x2 0x40000000 0x1 0x00000000>;
+	//};
+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		global_cma: linux,cma@58000000 {
+			compatible = "shared-dma-pool";
+			linux,cma-default;
+			reusable;
+			reg = <0x0 0x58000000 0x0 0x28000000>;
+		};
+
+		mmp_reserved: linux,multimedia {
+			compatible = "shared-dma-pool";
+			reusable;
+			reg = <0x00000000 0x80000000 0x0 0x10000000>;
+		};
+
+#if 0
+		isp_reserved: frame_buffer@90000000 {
+			compatible = "shared-dma-pool";
+			reusable;
+			reg = <0x0 0x90000000 0x0 0x18000000>;
+		};
+
+		opencva_reserved: OpenCVA@A8000000 {
+			reusable;
+			reg = <0x0 0xA8000000 0x0 0x7D00000>;
+		};
+
+		drp_codec: DRP-Codec@AFD00000 {
+			reusable;
+			reg = <0x0 0xAFD00000 0x0 0x300000>;
+		};
+
+		image_buf0: image_buf@B0000000 {
+			compatible = "shared-dma-pool";
+			reusable;
+			reg = <0x0 0xB0000000 0x0 0x4000000>;
+			label = "image_buf0";
+		};
+
+		cru_func: cru_buffer@B4000000 {
+			compatible = "shared-dma-pool";
+			reusable;
+			reg = <0x0 0xB4000000 0x0 0x16400000>;
+		};
+
+		drp_reserved: DRP-AI@240000000 {
+			reusable;
+			reg = <0x2 0x40000000 0x0 0x20000000>;
+		};
+#endif
+	};
+
+	mmngr {
+		compatible = "renesas,mmngr";
+		memory-region = <&mmp_reserved>;
+	};
+
+	mmngrbuf {
+		compatible = "renesas,mmngrbuf";
+	};
+
+	vspm_if {
+		compatible = "renesas,vspm_if";
+	};
+
+	reg_1p8v: regulator0 {
+		compatible = "regulator-fixed";
+
+		regulator-name = "fixed-1.8V";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+
+	reg_3p3v: regulator1 {
+		compatible = "regulator-fixed";
+
+		regulator-name = "fixed-3.3V";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+
+	vmmc_sdhi0: regulator-vcc-sdhi0 {
+		compatible = "regulator-fixed";
+
+		regulator-name = "SDHI0 Vmmc";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+
+		//gpios = <&pinctrl RZG2L_GPIO(10, 1) GPIO_ACTIVE_HIGH>;
+		//enable-active-high;
+	};
+
+	vccq_sdhi0: regulator-vccq-sdhi0 {
+		compatible = "regulator-gpio";
+
+		regulator-name = "SDHI0 VccQ";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+
+		//gpios = <&pinctrl RZG2L_GPIO(10, 0) GPIO_ACTIVE_HIGH>;
+
+		//gpios-states = <0>;
+		//states = <3300000 0 1800000 1>;
+	};
+
+	vmmc_sdhi1: regulator-vcc-sdhi1 {
+		compatible = "regulator-fixed";
+
+		regulator-name = "SDHI1 Vmmc";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+
+		//gpios = <&pinctrl RZG2L_GPIO(10, 3) GPIO_ACTIVE_HIGH>;
+		//enable-active-high;
+	};
+
+	vccq_sdhi1: regulator-vccq-sdhi1 {
+		compatible = "regulator-gpio";
+
+		regulator-name = "SDHI1 VccQ";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <3300000>;
+
+		//gpios = <&pinctrl RZG2L_GPIO(10, 2) GPIO_ACTIVE_HIGH>;
+
+		//gpios-states = <0>;
+		//states = <3300000 0 1800000 1>;
+	};
+
+#if 0
+	hdmi-out {
+		compatible = "hdmi-connector";
+		type = "d";
+
+		port {
+			hdmi_con_out: endpoint {
+				remote-endpoint = <&adv7535_out>;
+			};
+		};
+	};
+#endif
+
+	clk_ext_camera: clk_ext_camera {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <24000000>;
+	};
+
+	x1_clk: x1-clock {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <24000000>;
+	};
+};
+
+&audio_clka {
+	clock-frequency = <22579200>;
+};
+
+&audio_clkb {
+	clock-frequency = <24576000>;
+};
+
+&audio_clkc {
+	clock-frequency = <22579200>;
+};
+
+&pinctrl {
+	can0_pins: can0 {
+		pinmux = <RZG2L_PORT_PINMUX(8, 0, 5)>, /* TX */
+			 <RZG2L_PORT_PINMUX(8, 1, 5)>; /* RX */
+	};
+
+	can1_pins: can1 {
+		pinmux = <RZG2L_PORT_PINMUX(8, 2, 5)>, /* TX */
+			 <RZG2L_PORT_PINMUX(8, 3, 5)>; /* RX */
+	};
+
+	scif0_pins: scif0 {
+		pins =  "SCIF_RXD", "SCIF_TXD";
+	};
+
+	sci0_pins: sci0 {
+		pinmux = <RZG2L_PORT_PINMUX(9, 2, 5)>, /* SCI0_TXD_MOSI_SDA */
+			 <RZG2L_PORT_PINMUX(9, 3, 4)>; /* SCI0_RXD_MISO_SCL */
+	};
+
+	sci4_pins: sci4 {
+		pinmux = <RZG2L_PORT_PINMUX(8, 4, 6)>, /* SCI4_TXD_MOSI_SDA */
+			 <RZG2L_PORT_PINMUX(8, 5, 6)>, /* SCI4_RXD_MISO_SCL */
+			 <RZG2L_PORT_PINMUX(8, 0, 3)>, /* SCI4_CTS */
+			 <RZG2L_PORT_PINMUX(8, 1, 1)>; /* SCI4_RTS */
+	};
+
+	sci6_pins: sci6 {
+		pinmux = <RZG2L_PORT_PINMUX(9, 0, 2)>, /* SCI6_TXD_MOSI_SDA */
+			 <RZG2L_PORT_PINMUX(9, 1, 2)>; /* SCI6_RXD_MISO_SCL */
+	};
+
+	//sdhi0_pins: sd0 {
+	//	sd0_mux {
+	//		pinmux = <RZG2L_PORT_PINMUX(10, 5, 15)>; /* SD0_CD */
+	//	};
+	//};
+
+	sdhi1_pins: sd1 {
+		sd1_mux {
+			pinmux = <RZG2L_PORT_PINMUX(9, 4, 14)>; /* SD1_CD */
+		};
+	};
+
+	i2c0_pins: i2c0 {
+		pinmux = <RZG2L_PORT_PINMUX(3, 0, 1)>, /* I2C0_SDA */
+			 <RZG2L_PORT_PINMUX(3, 1, 1)>; /* I2C0_SCL */
+	};
+
+	i2c1_pins: i2c1 {
+		pinmux = <RZG2L_PORT_PINMUX(3, 2, 1)>, /* I2C1_SDA */
+			 <RZG2L_PORT_PINMUX(3, 3, 1)>; /* I2C1_SCL */
+	};
+
+	i2c2_pins: i2c2 {
+		pinmux = <RZG2L_PORT_PINMUX(2, 0, 4)>, /* I2C2_SDA */
+			 <RZG2L_PORT_PINMUX(2, 1, 4)>; /* I2C2_SCL */
+	};
+
+	i2c3_pins: i2c3 {
+		pinmux = <RZG2L_PORT_PINMUX(3, 6, 1)>, /* I2C3_SDA */
+			 <RZG2L_PORT_PINMUX(3, 7, 1)>; /* I2C3_SCL */
+	};
+
+	i2c4_pins: i2c4 {
+		pinmux = <RZG2L_PORT_PINMUX(4, 0, 1)>, /* I2C4_SDA */
+			 <RZG2L_PORT_PINMUX(4, 1, 1)>; /* I2C4_SCL */
+	};
+
+	i2c6_pins: i2c6 {
+		pinmux = <RZG2L_PORT_PINMUX(4, 4, 1)>, /* I2C6_SDA */
+			 <RZG2L_PORT_PINMUX(4, 5, 1)>; /* I2C6_SCL */
+	};
+
+	i2c8_pins: i2c8 {
+		pinmux = <RZG2L_PORT_PINMUX(0, 6, 1)>, /* I2C8_SDA */
+			 <RZG2L_PORT_PINMUX(0, 7, 1)>; /* I2C8_SCL */
+	};
+
+	usb30_pins: usb30 {
+		pinmux = <RZG2L_PORT_PINMUX(11, 0, 14)>; /* VBUS */
+	};
+
+	usb31_pins: usb31 {
+		pinmux = <RZG2L_PORT_PINMUX(6, 2, 14)>; /* VBUS */
+	};
+
+	usb20_pins: usb20 {
+		pinmux = <RZG2L_PORT_PINMUX(9, 5, 14)>, /* VBUS */
+			 <RZG2L_PORT_PINMUX(9, 6, 14)>; /* OVC */
+	};
+
+	usb21_pins: usb21 {
+		pinmux = <RZG2L_PORT_PINMUX(6, 6, 14)>, /* VBUS */
+			 <RZG2L_PORT_PINMUX(6, 7, 14)>; /* OVC */
+	};
+
+};
+
+&extal_clk {
+	clock-frequency = <24000000>;
+};
+
+&ostm2 {
+	status = "okay";
+};
+
+&ostm3 {
+	status = "okay";
+};
+
+&scif0 {
+	pinctrl-0 = <&scif0_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+};
+
+&sci0 {
+	pinctrl-0 = <&sci0_pins>;
+	pinctrl-names = "default";
+
+	//uart-has-rtscts;
+	status = "okay";
+};
+
+&sci4 {
+	pinctrl-0 = <&sci4_pins>;
+	pinctrl-names = "default";
+
+	uart-has-rtscts;
+	status = "okay";
+};
+
+&sci6 {
+	pinctrl-0 = <&sci6_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+/* eMMC on MSRZV2H */
+&sdhi0 {
+	//pinctrl-0 = <&sdhi0_pins>;
+	//pinctrl-names = "default";
+	vmmc-supply = <&vmmc_sdhi0>;
+	vqmmc-supply = <&vccq_sdhi0>;
+	bus-width = <8>;
+//	mmc-hs200-1_8v;
+	non-removable;
+	fixed-emmc-driver-type = <1>;
+	status = "okay";
+};
+
+/* SD card on MSRZV2H */
+&sdhi1 {
+	pinctrl-0 = <&sdhi1_pins>;
+	pinctrl-names = "default";
+	vmmc-supply = <&vmmc_sdhi1>;
+	vqmmc-supply = <&vccq_sdhi1>;
+	bus-width = <4>;
+	sd-uhs-sdr50;
+	sd-uhs-sdr104;
+	status = "okay";
+};
+
+&eth0 {
+	phy-handle = <&phy0>;
+	phy-mode = "rgmii-id";
+	status = "okay";
+
+	mdio {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		phy0: ethernet-phy@7 {
+			compatible = "ethernet-phy-id0022.1640",
+				     "ethernet-phy-ieee802.3-c22";
+			reg = <7>;
+			rxc-skew-psec = <1400>;
+			txc-skew-psec = <1400>;
+			rxdv-skew-psec = <0>;
+			txdv-skew-psec = <0>;
+			rxd0-skew-psec = <0>;
+			rxd1-skew-psec = <0>;
+			rxd2-skew-psec = <0>;
+			rxd3-skew-psec = <0>;
+			txd0-skew-psec = <0>;
+			txd1-skew-psec = <0>;
+			txd2-skew-psec = <0>;
+			txd3-skew-psec = <0>;
+		};
+	};
+};
+
+&eth1 {
+	phy-handle = <&phy1>;
+	phy-mode = "rgmii-id";
+	status = "okay";
+
+	mdio {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		phy1: ethernet-phy@7 {
+			compatible = "ethernet-phy-id0022.1640",
+				     "ethernet-phy-ieee802.3-c22";
+			reg = <7>;
+			rxc-skew-psec = <1400>;
+			txc-skew-psec = <1400>;
+			rxdv-skew-psec = <0>;
+			txdv-skew-psec = <0>;
+			rxd0-skew-psec = <0>;
+			rxd1-skew-psec = <0>;
+			rxd2-skew-psec = <0>;
+			rxd3-skew-psec = <0>;
+			txd0-skew-psec = <0>;
+			txd1-skew-psec = <0>;
+			txd2-skew-psec = <0>;
+			txd3-skew-psec = <0>;
+		};
+	};
+};
+
+&i2c0 {
+	pinctrl-0 = <&i2c0_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+
+};
+
+&i2c1 {
+	pinctrl-0 = <&i2c1_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+
+};
+
+&i2c2 {
+	pinctrl-0 = <&i2c2_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+};
+
+&i2c3 {
+	pinctrl-0 = <&i2c3_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+
+};
+
+&i2c6 {
+	pinctrl-0 = <&i2c6_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+
+};
+
+&i2c8 {
+	status = "okay";
+};
+
+&du {
+	status = "okay";
+};
+
+#if 0
+&dsi0 {
+	status = "okay";
+
+	ports {
+		port@1 {
+			dsi0_out: endpoint {
+				remote-endpoint = <&adv7535_in>;
+				data-lanes = <1 2 3 4>;
+			};
+		};
+	};
+};
+#endif
+
+#if 0
+&cru0 {
+	status = "okay";
+};
+
+&csi20 {
+	status = "okay";
+
+	ports {
+		port {
+			csi20_in: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				remote-endpoint = <&ov5645_to_csi20>;
+			};
+		};
+	};
+};
+
+&cru1 {
+	status = "okay";
+};
+
+&csi21 {
+	status = "okay";
+
+	ports {
+		port {
+			csi21_in: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				remote-endpoint = <&ov5645_to_csi21>;
+			};
+		};
+	};
+};
+
+&cru2 {
+	status = "okay";
+};
+
+&csi22 {
+	status = "okay";
+
+	ports {
+		port {
+			csi22_in: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				remote-endpoint = <&ov5645_to_csi22>;
+			};
+		};
+	};
+};
+
+&cru3 {
+	status = "okay";
+};
+
+&csi23 {
+	status = "okay";
+
+	ports {
+		port {
+			csi23_in: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				remote-endpoint = <&ov5645_to_csi23>;
+			};
+		};
+	};
+};
+#endif
+
+&xhci0 {
+		pinctrl-0 = <&usb30_pins>;
+		pinctrl-names = "default";
+
+		memory-region = <&global_cma>;
+		status = "okay";
+};
+
+&xhci1 {
+		pinctrl-0 = <&usb31_pins>;
+		pinctrl-names = "default";
+
+		memory-region = <&global_cma>;
+		status = "okay";
+};
+
+&usb20phyrst {
+	status = "okay";
+};
+
+&usb21phyrst {
+	status = "okay";
+};
+
+&ohci0 {
+		memory-region = <&global_cma>;
+
+		dr_mode = "otg";
+		status = "okay";
+};
+
+&ohci1 {
+		memory-region = <&global_cma>;
+
+		status = "okay";
+};
+
+&ehci0 {
+		memory-region = <&global_cma>;
+
+		dr_mode = "otg";
+		status = "okay";
+};
+
+&ehci1 {
+		memory-region = <&global_cma>;
+
+		status = "okay";
+};
+
+&hsusb {
+		dr_mode = "otg";
+		status = "okay";
+};
+
+&usb2_phy0 {
+		pinctrl-0 = <&usb20_pins>;
+		pinctrl-names = "default";
+
+		status = "okay";
+};
+
+&usb2_phy1 {
+		pinctrl-0 = <&usb21_pins>;
+		pinctrl-names = "default";
+
+		status = "okay";
+};
+
+&canfd {
+	pinctrl-0 = <&can0_pins &can1_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	channel0 {
+		status = "okay";
+	};
+
+	channel1 {
+		status = "okay";
+	};
+};
+
+&rtc {
+	status = "okay";
+};
+
+&wdt1 {
+	status = "okay";
+};
+
+#if 0
+&spi2 {
+	pinctrl-0 = <&spi2_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+};
+#endif
+
+&xspi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "micron,mt25qu512a", "jedec,spi-nor";
+		reg = <0>;
+		m25p,fast-read;
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <1>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@000000 {
+				label = "bl2";
+				reg = <0x00000000 0x0001D200>;
+				read-only;
+			};
+
+			partition@01D200 {
+				label = "fip";
+				reg = <0x0001D200 0x001C2E00>;
+				read-only;
+			};
+
+			partition@1E0000 {
+				label = "env";
+				reg = <0x001E0000 0x00020000>;
+				read-only;
+			};
+
+			partition@200000 {
+				label = "test-area";
+				reg = <0x00200000 0x00E00000>;
+			};
+		};
+	};
+};
+
+&pcie0 {
+	/* Map all possible DDR as inbound ranges */
+	dma-ranges = <0x42000000 0 0x40000000 0 0x40000000 0x4 0x00000000>;
+
+	status = "okay";
+};
+
+&adc {
+	status = "okay";
+};
+
+&cmtw0 {
+	status = "okay";
+};
+
+&cmtw1 {
+	status = "okay";
+};
diff --git a/arch/arm/mach-renesas/Kconfig.rcar3 b/arch/arm/mach-renesas/Kconfig.rcar3
index 6fd8db896c..f7243d9833 100644
--- a/arch/arm/mach-renesas/Kconfig.rcar3
+++ b/arch/arm/mach-renesas/Kconfig.rcar3
@@ -236,6 +236,11 @@ config TARGET_SMARC_RZG2UL
 	help
           Support for Renesas RZ/G2UL Dev Platform
 
+config TARGET_MSRZV2H_BERRY
+	bool "MSRZV2Hberry board"
+	help
+          Support for ARIES Embedded MSRZV2H module on Berry base board
+
 config TARGET_RZV2H_DEV
 	bool "RZ/V2H Dev board"
 	help
@@ -279,6 +284,7 @@ source "board/renesas/rzv2l-dev/Kconfig"
 source "board/renesas/rzg2lc-dev/Kconfig"
 source "board/renesas/rzg2ul-dev/Kconfig"
 source "board/renesas/rzv2h-dev/Kconfig"
+source "board/aries/msrzv2h/Kconfig"
 source "board/renesas/rzg3e-dev/Kconfig"
 
 config MULTI_DTB_FIT_UNCOMPRESS_SZ
diff --git a/board/aries/msrzv2h/Kconfig b/board/aries/msrzv2h/Kconfig
new file mode 100644
index 0000000000..e1688be321
--- /dev/null
+++ b/board/aries/msrzv2h/Kconfig
@@ -0,0 +1,15 @@
+if TARGET_MSRZV2H_BERRY
+
+config SYS_SOC
+	default "renesas"
+
+config SYS_BOARD
+	default "msrzv2h"
+
+config SYS_VENDOR
+	default "aries"
+
+config SYS_CONFIG_NAME
+	default "msrzv2hberry"
+
+endif
diff --git a/board/aries/msrzv2h/Makefile b/board/aries/msrzv2h/Makefile
new file mode 100644
index 0000000000..13a98dfeb2
--- /dev/null
+++ b/board/aries/msrzv2h/Makefile
@@ -0,0 +1,13 @@
+#
+# board/renesas/rzv2h-dev/Makefile
+#
+# Copyright (C) 2022 Renesas Electronics Corporation
+#
+# SPDX-License-Identifier: GPL-2.0+
+#
+
+ifdef CONFIG_SPL_BUILD
+obj-y	:= ../rcar-common/gen3-spl.o
+else
+obj-y	:= msrzv2h.o ../../renesas/rcar-common/v2h-common.o
+endif
diff --git a/board/aries/msrzv2h/msrzv2h.c b/board/aries/msrzv2h/msrzv2h.c
new file mode 100644
index 0000000000..65d12a45d8
--- /dev/null
+++ b/board/aries/msrzv2h/msrzv2h.c
@@ -0,0 +1,324 @@
+#include <common.h>
+#include <cpu_func.h>
+#include <image.h>
+#include <init.h>
+#include <malloc.h>
+#include <netdev.h>
+#include <dm.h>
+#include <dm/platform_data/serial_sh.h>
+#include <asm/processor.h>
+#include <asm/mach-types.h>
+#include <asm/io.h>
+#include <linux/bitops.h>
+#include <linux/errno.h>
+#include <asm/arch/sys_proto.h>
+#include <asm/gpio.h>
+#include <asm/arch/gpio.h>
+#include <asm/arch/renesas.h>
+#include <asm/arch/rcar-mstp.h>
+#include <asm/arch/sh_sdhi.h>
+#include <i2c.h>
+#include <mmc.h>
+#include <linux/delay.h>
+#include <efi_loader.h>
+
+DECLARE_GLOBAL_DATA_PTR;
+
+#define	PFC_BASE			0x10410000
+#define	CPG_BASE			0x10420000
+
+#define PWPR				(PFC_BASE + 0x3C04)
+
+#define	P_2A				(PFC_BASE + 0x002A)
+#define	PM_2A				(PFC_BASE + 0x0154)
+#define	PMC_2A				(PFC_BASE + 0x022A)
+#define PFC_PMC26			(PFC_BASE + 0x0226)
+#define PFC_PFC26			(PFC_BASE + 0x0498)
+#define PFC_PMC29			(PFC_BASE + 0x0229)
+#define PFC_PFC29			(PFC_BASE + 0x04A4)
+
+#define	PMC_20				(PFC_BASE + 0x0220)
+#define	PFC_20				(PFC_BASE + 0x0480)
+#define PMC_23				(PFC_BASE + 0x0223)
+#define PFC_23				(PFC_BASE + 0x048C)
+#define	PMC_24				(PFC_BASE + 0x0224)
+#define	PFC_24				(PFC_BASE + 0x0490)
+#define	PFC_OSCBYPS			(PFC_BASE + 0x3C00)
+
+#define PWPR_REGWE_A			BIT(6)
+#define	PWPR_REGWE_B			BIT(5)
+
+#define	CPG_CLKON_9			(CPG_BASE + 0x0624)
+#define	CPG_RST_9			(CPG_BASE + 0x0924)
+#define	CPG_RST_10			(CPG_BASE + 0x0928)
+
+/* CPG */
+#define CPG_BASE			0x10420000
+#define CPG_SSEL0			(CPG_BASE + 0x0300)
+#define CPG_SSEL1			(CPG_BASE + 0x0304)
+#define CPG_CLKON_11			(CPG_BASE + 0x062C)
+#define CPG_CLKON_12			(CPG_BASE + 0x0630)
+#define CPG_CLKMON_5			(CPG_BASE + 0x0814)
+#define CPG_CLKMON_6			(CPG_BASE + 0x0818)
+#define CPG_RST_11			(CPG_BASE + 0x092C)
+#define CPG_RSTMON_5			(CPG_BASE + 0x0A14)
+
+#define CPG_RST_USB			(CPG_BASE + 0x0928)
+#define CPG_RSTMON4_USB			(CPG_BASE + 0x0A10)
+#define CPG_RSTMON5_USB			(CPG_BASE + 0x0A14)
+#define CPG_CLKON_USB			(CPG_BASE + 0x062C)
+#define CPG_CLKMON_USB			(CPG_BASE + 0x0814)
+
+#define PFC_OEN				(PFC_BASE + 0x3C40)
+#define PFC_OEN_OEN0			BIT(0)
+#define PFC_OEN_OEN1			BIT(1)
+#define PFC_PWPR			(PFC_BASE + 0x3C04)
+
+#define ICU_IPTSR_REG				0x10400060
+
+/* USB */
+#define USBPHY20_BASE			(0x15830000)
+#define USBPHY21_BASE			(0x15840000)
+#define USBPHY20_RESET			(USBPHY20_BASE + 0x000u)
+#define USBPHY21_RESET			(USBPHY21_BASE + 0x000u)
+
+#define USB20_BASE			(0x15800000)
+#define USB21_BASE			(0x15810000)
+#define USBF_BASE			(0x15820000)
+
+#define COMMCTRL			0x800
+#define HcRhDescriptorA			0x048
+#define LPSTS				0x102
+#define USB2_PHY_UTMICTRL2		0xb04
+#define USB2_PHY_RESET			0x000
+#define USB2_PHY_OTGR			0x600
+
+#define SYS_ADC_CFG			0x10431600
+
+#if IS_ENABLED(CONFIG_EFI_HAVE_CAPSULE_SUPPORT)
+
+#define EFI_FIRMWARE_IMAGE_TYPE_RZV2H_GUID \
+	EFI_GUID(0x7f26b24e, 0x7cc4, 0x40a5, 0x8d, 0x3b, 0x0c, 0xbf, 0x47, 0x3f, 0x7a, 0x83)
+
+struct efi_fw_image fw_images[] = {
+	{
+		.image_type_id = EFI_FIRMWARE_IMAGE_TYPE_RZV2H_GUID,
+		.fw_name = u"RENESAS-FIP",
+		.image_index = 1,
+	},
+};
+
+struct efi_capsule_update_info update_info = {
+	.dfu_string = "sf 0:0=fip.bin raw 0x20000 0x1F0000\0",
+	.num_images = ARRAY_SIZE(fw_images),
+	.images = fw_images,
+};
+#endif /* EFI_HAVE_CAPSULE_SUPPORT */
+
+void s_init(void)
+{
+	*(volatile u32 *)PWPR |= (PWPR_REGWE_A | PWPR_REGWE_B);
+
+	/* Enable ADC */
+	*(volatile u32 *)(SYS_ADC_CFG) = 0;
+
+	/* SD1  */
+	*(volatile u8 *)PMC_2A   &= ~(0x03 << 2);/* PA3,PA2 port */
+	*(volatile u8 *)P_2A      = (*(volatile u32 *)P_2A  & ~(0x03<<2)) | (0x01 <<3); /* PA3=1,PA2=0		*/
+	*(volatile u16 *)PM_2A    = (*(volatile u32 *)PM_2A & ~(0x0f<<4)) | (0x0a <<4); /* PA3,PA2 output	*/
+
+	/* I2C3	*/
+	*(volatile u32 *)PFC_23  = (*(volatile u32 *)PFC_23 & 0x00FFFFFF) | (0x01 << 28) | (0x01 << 24);
+	*(volatile u8 *)PMC_23   |= (0x03) << 6;	/* P37,P36 multiplexed function	*/
+
+	/* I2C6	*/
+	*(volatile u32 *)PFC_24  = (*(volatile u32 *)PFC_24 & 0xFF00FFFF) | (0x01 << 20) | (0x01 << 16);
+	*(volatile u8 *)PMC_24   |= (0x03) << 4;	/* P45,P44 multiplexed function	*/
+
+	/* I2C3	*/
+	*(volatile u32 *)CPG_CLKON_9 = 0x00800080;
+	*(volatile u32 *)CPG_RST_9   = 0x08000800;
+
+	/* I2C6	*/
+	*(volatile u32 *)CPG_CLKON_9 = 0x04000400;
+	*(volatile u32 *)CPG_RST_9   = 0x40004000;
+
+	/* I2C8 */
+	*(volatile u32 *)PFC_20  = (*(volatile u32 *)PFC_20 & 0x00FFFFFF) | (0x01 << 28) | (0x01 << 24);
+	*(volatile u8 *)PMC_20   |= (0x03) << 6;	/* P07,P06 multiplexed function	*/
+
+	*(volatile u32 *)CPG_CLKON_9 = 0x00080008;
+	*(volatile u32 *)CPG_RST_10  = 0x00010001;
+
+	/* Enale OE of IO block for xSPI */
+	*(volatile u32 *)(PFC_OEN) &= ~GENMASK(5,2);
+
+	// Use PLL clock for clk_tx_i only for RGMII mode
+	// Wite OEN reg. OEN0 bit "0" for output direction
+	*(volatile u32 *)(PFC_OEN) &= ~(PFC_OEN_OEN1 | PFC_OEN_OEN0);
+	while((*(volatile u32 *)(PFC_OEN) & (PFC_OEN_OEN1 | PFC_OEN_OEN0)) != 0x0)
+		;
+	
+	*(volatile u32 *)PWPR &= ~(PWPR_REGWE_A | PWPR_REGWE_B);
+
+	/* Set Bypass and Powerdown mode for Audio OSC */
+	*(volatile u32 *)(PFC_OSCBYPS) = 0x001C0406;
+
+	*(volatile u32 *)(ICU_IPTSR_REG) = 0;
+	
+	/* Reset ETH 0,1 */
+	*(volatile u32 *)(CPG_RST_11) = 0x00030000;
+	while((*(volatile u32 *)(CPG_RSTMON_5) & 0x00000006) == 0x0)
+		;
+
+	/* Release reset ETH0,1 */
+	*(volatile u32 *)(CPG_RST_11) = 0x00030003;
+	while((*(volatile u32 *)(CPG_RSTMON_5) & 0x00000006) != 0x0)
+		;
+
+	/* Disable SMUX2_GBE0_RXCLK and SMUX2_GBE1_RXCLK */
+	*(volatile u32 *) (CPG_SSEL0) = 0x10000000;
+	*(volatile u32 *) (CPG_SSEL1) = 0x00100000;
+
+	/* Enable SMUX2_GBE0_RXCLK and SMUX2_GBE1_RXCLK */
+	*(volatile u32 *) (CPG_SSEL0) = 0x10001000;
+	*(volatile u32 *) (CPG_SSEL1) = 0x00100010;
+
+	/* Enable aclk_csr, aclk, tx, rx, tx_180, rx_180 for ETH0 */
+	/* Enable tx, rx for ETH1 */
+	*(volatile u32 *)(CPG_CLKON_11) = 0xFF00FF00;
+	while((*(volatile u32 *)(CPG_CLKMON_5) & 0xFF000000) != 0xFF000000)
+		;
+
+	/* Enable aclk_csr, aclk, tx_180, rx_180 for ETH1 */
+	*(volatile u32 *)(CPG_CLKON_12) = 0x000F000F;
+	while((*(volatile u32 *)(CPG_CLKMON_6) & 0x0000000F) != 0x0000000F)
+		;
+}
+
+static void _usbphy_init(void)
+{
+	/* Overwrite SLEEPM/SUSPENDM signals by USB2PHY Control */
+	(*(volatile u32 *)(USBPHY20_BASE + USB2_PHY_UTMICTRL2)) = 0x00000303;
+	(*(volatile u32 *)(USBPHY21_BASE + USB2_PHY_UTMICTRL2)) = 0x00000303;
+
+	/* Assert USB2PHY reset */
+	(*(volatile u32 *)(USBPHY20_BASE + USB2_PHY_RESET)) = 0x00000206;
+	(*(volatile u32 *)(USBPHY21_BASE + USB2_PHY_RESET)) = 0x00000206;
+
+	/* Delay 10us */
+	udelay(10);
+
+	/* De-Assert USB2PHY reset */
+	(*(volatile u32 *)(USBPHY20_BASE + USB2_PHY_RESET)) = 0x00000200;
+	(*(volatile u32 *)(USBPHY21_BASE + USB2_PHY_RESET)) = 0x00000200;
+
+	/* Release overwrites of SLEEPM/SUSMENDM signals, and RESET signal */
+	(*(volatile u32 *)(USBPHY20_BASE + USB2_PHY_UTMICTRL2)) = 0x00000003;
+	(*(volatile u32 *)(USBPHY20_BASE + USB2_PHY_RESET)) = 0;
+
+	(*(volatile u32 *)(USBPHY21_BASE + USB2_PHY_UTMICTRL2)) = 0x00000003;
+	(*(volatile u32 *)(USBPHY21_BASE + USB2_PHY_RESET)) = 0;
+
+	/* Activate VBUS Valid comparator */
+	(*(volatile u32 *)(USBPHY20_BASE + USB2_PHY_OTGR)) = 0x00000909;
+	(*(volatile u32 *)(USBPHY21_BASE + USB2_PHY_OTGR)) = 0x00000909;
+}
+
+static void _reset_usb2(void)
+{
+	/* Reset for USBTEST */
+	(*(volatile u32 *)CPG_RST_USB) |= 0x80008000;
+	while((*(volatile u32 *)(CPG_RSTMON5_USB) & 0x00000001) != 0x0);
+
+	/* Reset for USB2 host 0 and 1 */
+	(*(volatile u32 *)CPG_RST_USB) |= 0x30003000;
+	while((*(volatile u32 *)(CPG_RSTMON4_USB) & 0x60000000) != 0x0);
+}
+
+static void board_usb_init(void)
+{
+	/* Reset USB*/
+	_reset_usb2();
+
+	/* Enable clock for USB */
+	(*(volatile u32 *)CPG_CLKON_USB) = 0x00F800F8;
+	while((*(volatile u32 *)(CPG_CLKMON_USB) & 0x00F80000) != 0x00F80000);
+
+	/* Setup  */
+	/* Disable GPIO Write Protect */
+	(*(volatile u32 *)PFC_PWPR) |= (0x1u << 6);
+
+        /* Set P9_5 as Func.14 for VBUSEN */
+        /* Control mode (multiplexed function) */
+        (*(volatile u32 *)PFC_PMC29) |= (0x1u << 5);
+        (*(volatile u32 *)PFC_PFC29) &= ~(0xF << 20);
+        /* Function mode 15 */
+        (*(volatile u32 *)PFC_PFC29) |= (0x0E << 20);
+
+        /* Set P9_6 as Func.14 for OVRCUR */
+        /* Control mode (multiplexed function) */
+        (*(volatile u32 *)PFC_PMC29) |= (0x1u << 6);
+        (*(volatile u32 *)PFC_PFC29) &= ~(0xF << 24);
+        /* Function mode 14 */
+        (*(volatile u32 *)PFC_PFC29) |= (0x0E << 24);
+
+        /* Set P6_6 as Func.14 for VBUSEN */
+        /* Control mode (multiplexed function) */
+         (*(volatile u32 *)PFC_PMC26) |= (0x1u << 6);
+         (*(volatile u32 *)PFC_PFC26) &= ~(0xF << 24);
+        /* Function mode 14 */
+         (*(volatile u32 *)PFC_PFC26) |= (0x0E << 24);
+
+        /* Set P6_7 as Func.14 for OVRCUR */
+        /* Control mode (multiplexed function) */
+        (*(volatile u32 *)PFC_PMC26) |= (0x1u << 7);
+        (*(volatile u32 *)PFC_PFC26) &= ~(0xF << 28);
+        /* Function mode 15 */
+        (*(volatile u32 *)PFC_PFC26) |= (0x0E << 28);
+
+	/* Enable Write protect */
+	(*(volatile u32 *)PFC_PWPR) &= ~(0x1u << 6);
+
+	/* Initialize phy */
+	_usbphy_init();
+
+	/*USB0 is HOST*/
+	(*(volatile u32 *)(USB20_BASE + COMMCTRL)) = 0;
+
+	/*USB1 is HOST*/
+	(*(volatile u32 *)(USB21_BASE + COMMCTRL)) = 0;
+
+	/* Set USBPHY normal operation (Function only) */
+	(*(volatile u16 *)(USBF_BASE + LPSTS)) |= (0x1u << 14);
+
+	/* Overcurrent is not supported */
+	(*(volatile u32 *)(USB20_BASE + HcRhDescriptorA)) |= (0x1u << 12);
+	(*(volatile u32 *)(USB21_BASE + HcRhDescriptorA)) |= (0x1u << 12);
+}
+
+int board_late_init(void)
+{
+	return 0;
+}
+
+int board_early_init_f(void)
+{
+
+	return 0;
+}
+
+int board_init(void)
+{
+	/* adress of boot parameters */
+	gd->bd->bi_boot_params = CONFIG_TEXT_BASE + 0x50000;
+
+	board_usb_init();
+
+	return 0;
+}
+
+void reset_cpu(void)
+{
+
+}
diff --git a/configs/msrzv2hberry_defconfig b/configs/msrzv2hberry_defconfig
new file mode 100644
index 0000000000..75a87f0681
--- /dev/null
+++ b/configs/msrzv2hberry_defconfig
@@ -0,0 +1,189 @@
+CONFIG_USB_GADGET=y
+CONFIG_USB_RENESAS_USBHS=y
+CONFIG_USB_GADGET_DOWNLOAD=y
+CONFIG_USB_FUNCTION_MASS_STORAGE=y
+CONFIG_CMD_USB_MASS_STORAGE=y
+CONFIG_ARM=y
+CONFIG_ARCH_CPU_INIT=y
+CONFIG_ARCH_RENESAS=y
+CONFIG_OF_UPSTREAM=n
+CONFIG_MTD=y
+CONFIG_TEXT_BASE=0x50000000
+CONFIG_ENV_SIZE=0x20000
+CONFIG_ENV_OFFSET=0x001E0000
+CONFIG_SYS_LOAD_ADDR=0x58000000
+CONFIG_DM_GPIO=y
+CONFIG_RCAR_GEN3=y
+CONFIG_R9A09G057=y
+CONFIG_TARGET_MSRZV2H_BERRY=y
+CONFIG_DEFAULT_DEVICE_TREE="msrzv2hberry"
+CONFIG_SMBIOS_PRODUCT_NAME=""
+CONFIG_SPL=n
+CONFIG_FIT=y
+CONFIG_SUPPORT_RAW_INITRD=y
+CONFIG_DEFAULT_FDT_FILE="msrzv2hberry.dtb"
+CONFIG_VERSION_VARIABLE=y
+CONFIG_HUSH_PARSER=y
+CONFIG_CMD_BOOTZ=y
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_I2C=y
+CONFIG_CMD_MMC=y
+CONFIG_CMD_PCI=y
+CONFIG_CMD_USB=y
+CONFIG_CMD_DHCP=y
+CONFIG_CMD_MII=y
+CONFIG_CMD_PING=y
+CONFIG_CMD_EXT2=y
+CONFIG_CMD_EXT4=y
+CONFIG_CMD_EXT4_WRITE=y
+CONFIG_CMD_FAT=y
+CONFIG_CMD_FS_GENERIC=y
+CONFIG_OF_CONTROL=y
+CONFIG_OF_PLATDATA=y
+CONFIG_OF_LIST="msrzv2hberry"
+CONFIG_MULTI_DTB_FIT_LZO=y
+CONFIG_MULTI_DTB_FIT_USER_DEFINED_AREA=y
+CONFIG_ENV_OVERWRITE=y
+CONFIG_SYS_RELOC_GD_ENV_ADDR=y
+CONFIG_REGMAP=y
+CONFIG_SYSCON=y
+CONFIG_CLK=y
+CONFIG_CLK_RENESAS=y
+CONFIG_CLK_R9A09G057=y
+CONFIG_RCAR_GPIO=y
+#CONFIG_PINCTRL_PFC_RZG2L=y
+CONFIG_PINCTRL_PFC_RZV2H=y
+CONFIG_RZV2H_GPIO=y
+CONFIG_DM_I2C=y
+CONFIG_SYS_I2C_RCAR_IIC=y
+CONFIG_SH_SDHI=y
+CONFIG_BITBANGMII=y
+CONFIG_PHY_MICREL=y
+CONFIG_PHY_MICREL_KSZ90X1=y
+CONFIG_NET_RANDOM_ETHADDR=y
+CONFIG_DM_ETH=y
+CONFIG_DM_ETH_PHY=y
+CONFIG_DWC_ETH_QOS=y
+CONFIG_DWC_ETH_QOS_RZV2H=y
+CONFIG_RENESAS_RAVB=y
+CONFIG_PCI=y
+CONFIG_DM_PCI=y
+CONFIG_PCI_RCAR_GEN3=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_REGULATOR_GPIO=y
+CONFIG_SCIF_CONSOLE=y
+CONFIG_TEE=y
+CONFIG_OPTEE=y
+CONFIG_USB=y
+CONFIG_USB_OTG=y
+CONFIG_DM_USB=y
+CONFIG_USB_XHCI_HCD=y
+CONFIG_USB_EHCI_HCD=y
+CONFIG_USB_EHCI_GENERIC=y
+CONFIG_USB_STORAGE=y
+CONFIG_OF_LIBFDT_OVERLAY=y
+CONFIG_SMBIOS_MANUFACTURER=""
+CONFIG_ENV_IS_IN_SPI_FLASH=y
+CONFIG_SYS_I2C_RZG2L_RIIC=y
+CONFIG_SPI_FLASH=y
+CONFIG_SPI_FLASH_BAR=y
+CONFIG_SPI_FLASH_USE_4K_SECTORS=y
+CONFIG_SPI=y
+CONFIG_DM_SPI=y
+CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_RENESAS_RPC_SPI=y
+CONFIG_CMD_SF=y
+CONFIG_CMD_SPI=y
+CONFIG_SPI_FLASH_STMICRO=y
+CONFIG_DM_SPI_FLASH=y
+CONFIG_ENV_SECT_SIZE=0x20000
+#Core UEFI features:
+CONFIG_BOOTM_EFI=y
+CONFIG_CMD_BOOTEFI=y
+CONFIG_CMD_NVEDIT_EFI=y
+CONFIG_CMD_EFIDEBUG=y
+CONFIG_CMD_BOOTEFI_HELLO=y
+CONFIG_CMD_BOOTEFI_HELLO_COMPILE=y
+CONFIG_CMD_BOOTEFI_SELFTEST=y
+CONFIG_CMD_GPT=y
+CONFIG_EFI_PARTITION=y
+CONFIG_EFI_LOADER=y
+CONFIG_EFI_DEVICE_PATH_TO_TEXT=y
+CONFIG_EFI_UNICODE_COLLATION_PROTOCOL2=y
+CONFIG_EFI_UNICODE_CAPITALIZATION=y
+CONFIG_EFI_HAVE_RUNTIME_RESET=y
+CONFIG_CMD_EFI_VARIABLE_FILE_STORE=y
+#Real Time Clock (RTC) support:
+CONFIG_DM_RTC=y
+CONFIG_EFI_GET_TIME=y
+CONFIG_EFI_SET_TIME=y
+CONFIG_RTC_EMULATION=y
+#update firmware:
+CONFIG_CMD_DFU=y
+CONFIG_FLASH_CFI_MTD=y
+CONFIG_EFI_CAPSULE_FIRMWARE_FIT=y
+CONFIG_EFI_CAPSULE_FIRMWARE_MANAGEMENT=y
+CONFIG_EFI_CAPSULE_FIRMWARE=y
+CONFIG_EFI_CAPSULE_FIRMWARE_RAW=y
+CONFIG_EFI_CAPSULE_FMP_HEADER=y
+# to enable partitions and filesystems support:
+CONFIG_FAT_WRITE=y
+CONFIG_FS_FAT=y
+CONFIG_CMD_PART=y
+CONFIG_PARTITIONS=y
+CONFIG_DOS_PARTITION=y
+CONFIG_ISO_PARTITION=y
+CONFIG_PARTITION_UUIDS=y
+#Device Firmware update:
+CONFIG_OF_LIBFDT=y
+#CONFIG_DISTRO_DEFAULTS=y
+CONFIG_CMD_RTC=y
+CONFIG_EFI_RUNTIME_UPDATE_CAPSULE=y
+CONFIG_EFI_HAVE_CAPSULE_SUPPORT=y
+CONFIG_EFI_CAPSULE_ON_DISK=y
+CONFIG_SPL_DFU=y
+CONFIG_DM_MTD=y
+CONFIG_DFU_MTD=y
+CONFIG_SPL_DFU_MTD=y
+CONFIG_DFU_SF=y
+CONFIG_DFU_OVER_USB=y
+CONFIG_DFU=y
+CONFIG_DFU_MMC=y
+CONFIG_DFU_NAND=y
+CONFIG_DFU_RAM=y
+CONFIG_DFU_SF_PART=y
+CONFIG_DFU_TIMEOUT=y
+CONFIG_DFU_OVER_TFTP=y
+CONFIG_DFU_TFTP=y
+#generate capsule files
+#CONFIG_EFI_CAPSULE_AUTHENTICATE=y
+CONFIG_EFI_SECURE_BOOT=y
+CONFIG_EFI_SIGNATURE_SUPPORT=y
+CONFIG_RSA=y
+CONFIG_RSA_VERIFY=y
+CONFIG_RSA_VERIFY_WITH_PKEY=y
+CONFIG_IMAGE_SIGN_INFO=y
+CONFIG_RSA_SOFTWARE_EXP=y
+CONFIG_ASYMMETRIC_KEY_TYPE=y
+CONFIG_ASYMMETRIC_PUBLIC_KEY_SUBTYPE=y
+CONFIG_RSA_PUBLIC_KEY_PARSER=y
+CONFIG_X509_CERTIFICATE_PARSER=y
+CONFIG_PKCS7_MESSAGE_PARSER=y
+CONFIG_PKCS7_VERIFY=y
+CONFIG_HASH=y
+CONFIG_SHA1=y
+CONFIG_SHA256=y
+CONFIG_SHA512=y
+CONFIG_SHA384=y
+CONFIG_MD5=y
+CONFIG_CRC32=y
+#EFI System Resource Table (ESRT)
+CONFIG_EFI_ESRT=y
+#Secure boot
+CONFIG_FIT_SIGNATURE=y
+
+CONFIG_PROT_TCP=y
+CONFIG_PROT_TCP_SACK=y
+CONFIG_CMD_WGET=y
+CONFIG_BOARD_LATE_INIT=y
diff --git a/include/configs/msrzv2hberry.h b/include/configs/msrzv2hberry.h
new file mode 100644
index 0000000000..c775c3c416
--- /dev/null
+++ b/include/configs/msrzv2hberry.h
@@ -0,0 +1,68 @@
+/* SPDX-License-Identifier: GPL-2.0+ */
+/*
+ * Copyright (C) 2022 Renesas Electronics Corporation
+ */
+
+#ifndef __MSRZV2HBERRY_H
+#define __MSRZV2HBERRY_H
+
+#include <asm/arch/renesas.h>
+
+#define CONFIG_REMAKE_ELF
+
+#ifdef CONFIG_SPL
+#define CONFIG_SPL_TARGET	"spl/u-boot-spl.scif"
+#endif
+
+/* boot option */
+
+#define CONFIG_CMDLINE_TAG
+#define CONFIG_SETUP_MEMORY_TAGS
+#define CONFIG_INITRD_TAG
+
+/* Generic Interrupt Controller Definitions */
+/* RZ/V2H use GIC-v3 */
+#define CONFIG_GICV3
+#define GICD_BASE	0x14900000
+#define GICR_BASE	0x14940000
+
+/* console */
+#define CONFIG_SYS_CBSIZE		2048
+#define CONFIG_SYS_BARGSIZE		CONFIG_SYS_CBSIZE
+#define CONFIG_SYS_BAUDRATE_TABLE	{ 115200, 38400 }
+
+/* PHY needs a longer autoneg timeout */
+#define PHY_ANEG_TIMEOUT		20000
+
+/* MEMORY */
+#define CONFIG_SYS_INIT_SP_ADDR		CONFIG_SYS_TEXT_BASE
+
+#define	DRAM_RSV_SIZE			0x08000000
+#define	CFG_MAX_MEM_MAPPED		(0x80000000u - DRAM_RSV_SIZE)
+
+/* SDHI clock freq */
+#define CONFIG_SH_SDHI_FREQ		133000000
+
+/* The HF/QSPI layout permits up to 1 MiB large bootloader blob */
+#define CONFIG_BOARD_SIZE_LIMIT		1048576
+
+/* ENV setting */
+#define CFG_EXTRA_ENV_SETTINGS	\
+	"usb_pgood_delay=2000\0"	\
+	"bootm_size=0x10000000\0"	\
+	"prodsdbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk1p2 \0" \
+	"prodemmcbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk0p2 \0" \
+	"bootimage=booti 0x48080000 - 0x48000000 \0" \
+	"fdt_file=boot/msrzv2hberry.dtb \0" \
+	"image_file=boot/Image \0" \
+	"emmcload=ext4load mmc 0:2 0x48080000 ${image_file];ext4load mmc 0:2 0x48000000 ${fdt_file};run prodemmcbootargs \0" \
+	"sdload=ext4load mmc 1:2 0x48080000 ${image_file];ext4load mmc 1:2 0x48000000 ${fdt_file};run prodsdbootargs \0" \
+	"bootcmd_check=if mmc dev 1; then run sdload; else run emmcload; fi \0"
+
+#define CONFIG_BOOTCOMMAND	"run bootcmd_check bootimage"
+
+/* For board */
+/* Ethernet RAVB */
+#define CONFIG_BITBANGMII_MULTI
+
+#endif /* __MSRZV2HBERRY_H */
-- 
2.34.1

